### PR TITLE
Migrate to Python3 + misc improvements

### PR DESCRIPTION
**Pull request by scriptSQD**
Hope this will get merged somehow. Comments appreciated.

If I removed something mandatory, let me know and we'll fix that. Meanwhile, I'll keep trying to get Wayland to work.

### CODE REVIEW

**Review by Justicechukwunweike2022 — APPROVED** [review]

**Review by Justicechukwunweike2022 — APPROVED** [review]

File: installer/linux/debian_package/DEBIAN/postinst [hunk 1]
```diff
@@ -2,8 +2,21 @@
 # This file will be overwritten by rpm specific script which is passed to fpm in build script
 # So if multi-platform change is needed remember to change that file as well
 set -e
+
+# Determine is system uses systemd as init system
```
Comment (milahu) [inline installer/linux/debian_package/DEBIAN/postinst:6]: typo: is -> if

File: installer/linux/debian_package/DEBIAN/preinst [hunk 2]
```diff
@@ -4,27 +4,46 @@ end=$((SECONDS+10))
...
+else
+echo "Unable to determine init system"
+exit 1
+fi
 
 # Stop and disable firewalld service if it is active, otherwise it will conflict with iptables on OS reboot
 # Relevant for Fedora distributions
-systemctl is-active --quiet firewalld
-if [ $? -eq 0 ] 
+if [ -f "/usr/sbin/firewalld" ]
```
Comment (milahu) [inline installer/linux/debian_package/DEBIAN/preinst:35]: not portable. use `command -v firewalld`

File: installer/linux/additional_files/postinst_rpm [hunk 3]
```diff
@@ -2,8 +2,21 @@
 # This file overwrites the debian-specific postinst during fpm call in build script 
 # So if multi-platform change is needed remember to change that file as well
 set -e
+
+# Determine is system uses systemd as init system
+if [ -d "/run/systemd/system" ]
```
Comment (milahu) [inline installer/linux/additional_files/postinst_rpm:7]: not portable. use `command -v systemctl`

File: installer/linux/additional_files/postinst_rpm [hunk 4]
```diff
@@ -2,8 +2,21 @@
...
+then
 systemctl enable windscribe-helper
 systemctl start windscribe-helper
+elif [ -d "/run/openrc" ]
```
Comment (milahu) [inline installer/linux/additional_files/postinst_rpm:11]: not portable. use `command -v rc-update`

File: tools/base/extract.py [hunk 5]
```diff
@@ -7,91 +7,91 @@
...
 
 class Extractor:
-    # memoize so we don't have to read the file more than once per variable
-    extracted_app_version = ""
-    extracted_app_version_with_beta_suffix = ""
-    extracted_mac_signing_params = ("", "")
+	# memoize so we don't have to read the file more than once per variable
+	extracted_app_version = ""
+	extracted_app_version_with_beta_suffix = ""
+	extracted_mac_signing_params = ("", "")
```
Comment (milahu) [inline tools/base/extract.py:17]: keep indent to avoid diff noise

File: .gitignore [hunk 6]
```diff
@@ -25,4 +25,8 @@ common/keys/linux/key.pem
 common/keys/linux/key.pub
 common/keys/linux/key_pub.txt
 installer/windows/signing/code_signing.pfx
-backend/mac/provisioning_profile/
\ No newline at end of file
+backend/mac/provisioning_profile/
+
+# IDEs / Editors
```
Comment (milahu) [inline .gitignore:30]: split changes into multiple commits

- gitignore
- python3
- system-v (rc-update)

File: README.md [hunk 7]
```diff
@@ -1,116 +1,133 @@
...
-- CppCheck (run install_cppcheck.bat with admin rights from common/prepare_build_environment/windows).
-- Verify the following entries are in your System PATH environment variable. If they are not, add them to the System PATH environment variable.  Reboot.
-    - C:\Python27
-    - C:\Python27\Scripts
-    - C:\Perl64\site\bin (or equivalent Strawberry Perl site\bin folder)
-    - C:\Perl64\bin (or equivalent Strawberry Perl bin folder)
-    - C:\Program Files\Git\cmd
-    - C:\Program Files\Cppcheck
+-   Windows 10.
+-   Install git (https://git-scm.com/downloads). When installing Git, you can stick with all the default options presented to you by the installer.
```
Comment (milahu) [inline README.md:19]: keep whitespace

File: tools/base/utils.py [hunk 8]
```diff
@@ -4,15 +4,15 @@
...
+import glob
 import platform
 import os
 import shutil
 import stat
 import time
 
 import yaml
-import messages as msg
+import base.messages as msg
```
Comment (milahu) [inline tools/base/utils.py:15]: from . import messages as msg

**Comment by milahu** [comment]
migrating to python3 is as simple as

```
2to3 -w -n .
```

but the build system is just a complex wrapper for qmake, which is deprecated for cmake ...

**Comment by bernerdad** [comment]
We've migrated to python3 for the latest (v2.6) release. Thanks so much for the PR!

**Pull request closed without merging**